src/main/java/tree/ImpTargetTreeNode.kt

```diff
@@ -1,30 +1,100 @@
 package tree
 
-class ImpTargetTreeNode : TargetTreeNode {
-    override fun moveRight(): TargetTreeNode {
-        println("moveRight")
-        return this
+class ImpTargetTreeNode() : TargetTreeNode {
+
+    private var name: String = ""
+    private var tag: String = ""
+    private var id: Int = -1
+    private var children: List<TargetTreeNode>? = null
+    private var parent: TargetTreeNode? = null
+
+    constructor(name: String, tag: String, id: Int) : this() {
+        this.name = name
+        this.tag = tag
+        this.id = id
+    }
+
+    override fun moveToRightSibling(): TargetTreeNode? {
+        if (parent != null)
+            return (parent!! as ImpTargetTreeNode).getRightSibling(this)
+        return null
```

Comment:
I think we can refactor this to:

```suggestion
        val rightSibling = parent as? ImpTargetTreeNode
        return rightSibling?.getRightSibling(this) 
```
This will return `null` if `rightSibling` is `null` or `.getRightSibling(this)` if is not null.

This might also work for the bellow methods.

Reply:
Wow, nice find! I'll leave your suggestion as is.